include/exadg/time_integration/restart_data.h

```diff
@@ -99,6 +108,33 @@ struct RestartData
 
   // counter needed do decide when to write restart
   mutable unsigned int counter;
+
+  // Finite element degree used when restart data was written (relevant for restart run only).
+  unsigned int degree_u;
+  unsigned int degree_p;
+
+  // TriangulationType used when restart data was written (relevant for restart run only).
+  TriangulationType triangulation_type;
+
+  // The discretization used when writing the restart data was identical to the current one
+  // (after calling `ApplicationBase::create_grid()`). Note that this includes the finite
+  // element, uniform and adaptive refinement, and the TriangulationType, *but* one might
+  // consider a different number of MPI ranks for `dealii::parallel::distributed::Triangulation``
+  // without the need for the otherwise necessary global projection.
+  bool discretization_identical;
+
+  // The mapping of the triangulation should be de-/serialized as well to consider for a mapped
+  // geometry at serialization and during deserialization. This is option toggles storing the
+  // mapping via a displacement vector *and* reading it back in. Hence, this parameter needs to
+  // match in serialization/deserialization runs.
+  bool consider_mapping;
+
+  // The `mapping_degree` considered when storing or reading the grid.
+  unsigned int mapping_degree;
+
+  // Parameters for `dealii::RemotePointEvaluation` used for grid-to-grid projection.
+  double rpe_tolerance_unit_cell;
+  bool   rpe_enforce_unique_mapping;
```

Comment:
From a design perspective, RestartData should contain an object of a class like `GridToGridProjectionData` (to be introduced), which can be extended, e.g. by solver tolerance parameters, etc.